Jenkinsfile

```diff
@@ -96,36 +97,15 @@ def job = {
         },
         downstreamBuildsStepName: {
             echo "Building cp-downstream-builds"
-            if (config.isPrJob) {
-                try {
-                    def muckrakeBranch = kafkaMuckrakeVersionMap[env.CHANGE_TARGET]
-                    def forkRepo = "${env.CHANGE_FORK ?: "confluentinc"}/kafka.git"
-                    def forkBranch = env.CHANGE_BRANCH
-                    echo "Schedule test-cp-downstream-builds with :"
-                    echo "Muckrake branch : ${muckrakeBranch}"
-                    echo "PR fork repo : ${forkRepo}"
-                    echo "PR fork branch : ${forkBranch}"
-                    buildResult = build job: 'test-cp-downstream-builds', parameters: [
-                            [$class: 'StringParameterValue', name: 'BRANCH', value: muckrakeBranch],
-                            [$class: 'StringParameterValue', name: 'TEST_PATH', value: "muckrake/tests/dummy_test.py"],
-                            [$class: 'StringParameterValue', name: 'KAFKA_REPO', value: forkRepo],
-                            [$class: 'StringParameterValue', name: 'KAFKA_BRANCH', value: forkBranch]],
-                            propagate: true, wait: true
-                    downstreamBuildFailureOutput = "cp-downstream-builds result: " + buildResult.getResult();
-                    return downstreamBuildFailureOutput
-                } catch (Exception e) {
-                    currentBuild.result = 'UNSTABLE'
-                    downstreamBuildFailureOutput = "cp-downstream-builds result: " + e.getMessage()
-                    writeFile file: "downstream/cp-downstream-build-failure.txt", text: downstreamBuildFailureOutput
-                    archiveArtifacts artifacts: 'downstream/*.txt'
-
-                    return downstreamBuildFailureOutput
+            stage('Downstream validation') {
+                if (config.isPrJob && config.downStreamValidate) {
+                    downStreamValidation(true, true)
```

Comment:
You mentioned that we make it obvious that the downstream validation failed now. Out of interest, how is this shown in Jenkins? The previous approach was hacky (capture a txt file, etc), so it'll be nice if the new approach is better.

Reply:
Some screen shots can be find here https://confluentinc.atlassian.net/wiki/spaces/TOOLS/pages/1179166502/Dependency-mgmt+FAQ, And one example use common https://jenkins.confluent.io/job/confluentinc-pr/job/common/job/PR-255/

Reply:
Thanks! I assume the new downstreamValidation flag signals whether to mark the build as unstable or not? If so, LGTM.

Reply:
Yes, I add an option to mark the build as `UNSTABLE`, I want to introduce less custom code in individual build files but give more option in the library.  https://github.com/confluentinc/jenkins-common/blob/master/vars/downStreamValidation.groovy#L93-L96